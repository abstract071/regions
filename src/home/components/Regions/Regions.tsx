import React, {
  useEffect
} from 'react'
import {
  useSelector,
  useDispatch
} from 'react-redux'
import Grid from '@material-ui/core/Grid'
import CircularProgress from '@material-ui/core/CircularProgress'

import SimpleList from '../../../common/components/SimpleList'

import {
  regionsSelector
} from '../../../selectors'

import {
  getRegions
} from '../../../actions'


const Regions: React.FC = () => {
  const {
    // error,
    isLoading,
    data: regions
  } = useSelector( regionsSelector )
  const dispatch = useDispatch()

  useEffect( () => {
    dispatch( getRegions() )
  }, [] )

  return (
    <Grid className="hrz-center" item xs sm={ 10 } md={ 6 }>
      {
        !isLoading ? (
          <SimpleList
            values={ regions }
          />
        ) : <CircularProgress />
      }
    </Grid>
  )
}

export default Regions
