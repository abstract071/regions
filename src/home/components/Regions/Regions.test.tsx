import React from 'react'
import { shallow } from 'enzyme'

import Regions from './Regions'


describe( 'Regions', () => {
  let shallowedComponent: any = null

  beforeEach( () => {
    shallowedComponent = shallow(
      <Regions />
    )
  } )

  it( 'should match snapshot', () => {
    expect( shallowedComponent ).toMatchSnapshot()
  } )
} )
