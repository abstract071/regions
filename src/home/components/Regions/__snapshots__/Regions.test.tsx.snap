// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regions should match snapshot 1`] = `ShallowWrapper {}`;
